pkg/openstack/tasks/servers.go

```diff
@@ -0,0 +1,242 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package tasks
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/openstack/compute/v2/servers"
+	"github.com/gophercloud/gophercloud/v2/pagination"
+	"github.com/hibiken/asynq"
+
+	asynqclient "github.com/gardener/inventory/pkg/clients/asynq"
+	"github.com/gardener/inventory/pkg/clients/db"
+	openstackclients "github.com/gardener/inventory/pkg/clients/openstack"
+	"github.com/gardener/inventory/pkg/openstack/models"
+	asynqutils "github.com/gardener/inventory/pkg/utils/asynq"
+)
+
+const (
+	// TaskCollectServers is the name of the task for collecting OpenStack
+	// servers.
+	TaskCollectServers = "openstack:task:collect-servers"
+)
+
+// CollectServersPayload represents the payload, which specifies
+// where to collect OpenStack Servers from.
+type CollectServersPayload struct {
+	// Project specifies the project from which to collect.
+	ProjectID string `json:"project_id" yaml:"project_id"`
+}
+
+// NewCollectServersTask creates a new [asynq.Task] for collecting OpenStack
+// servers, without specifying a payload.
+func NewCollectServersTask() *asynq.Task {
+	return asynq.NewTask(TaskCollectServers, nil)
+}
+
+// HandleCollectServersTask handles the task for collecting OpenStack Servers.
+func HandleCollectServersTask(ctx context.Context, t *asynq.Task) error {
+	// If we were called without a payload, then we enqueue tasks for
+	// collecting OpenStack Servers from all known projects.
+	data := t.Payload()
+	if data == nil {
+		return enqueueCollectServers(ctx)
+	}
+
+	var payload CollectServersPayload
+	if err := asynqutils.Unmarshal(data, &payload); err != nil {
+		return asynqutils.SkipRetry(err)
+	}
+
+	if payload.ProjectID == "" {
+		return asynqutils.SkipRetry(errors.New("no project ID specified"))
```

Comment:
This error would most likely be common to all OpenStack related handlers, so it's better to make a package-scoped var for it.